From a8a79a16cc3e895135f3f2860270f35c7d459611 Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Thu, 2 Oct 2014 14:16:43 +0100
Subject: [PATCH 332/351] ASoC: florida: Add delay after SYSCLK patch

When updating the write sequencer we must ensure there is a small
delay before anything uses the write sequencer after the patch has
been applied.

Change-Id: Ia643033595a07cff3c4a90e526e75180b6e6d67f
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
---
 sound/soc/codecs/arizona.c |    2 ++
 sound/soc/codecs/florida.c |    2 ++
 2 files changed, 4 insertions(+)

diff --git a/sound/soc/codecs/arizona.c b/sound/soc/codecs/arizona.c
index 26bc464..438eeca 100644
--- a/sound/soc/codecs/arizona.c
+++ b/sound/soc/codecs/arizona.c
@@ -1056,6 +1056,7 @@ static int florida_hp_pre_enable(struct snd_soc_dapm_widget *w)
 					       florida_no_dre_left_enable,
 					       ARRAY_SIZE(florida_no_dre_left_enable));
 		}
+		udelay(1000);
 		break;
 	case ARIZONA_OUT1R_ENA_SHIFT:
 		if (val & ARIZONA_DRE1R_ENA_MASK) {
@@ -1067,6 +1068,7 @@ static int florida_hp_pre_enable(struct snd_soc_dapm_widget *w)
 					       florida_no_dre_right_enable,
 					       ARRAY_SIZE(florida_no_dre_right_enable));
 		}
+		udelay(1000);
 		break;
 
 	default:
diff --git a/sound/soc/codecs/florida.c b/sound/soc/codecs/florida.c
index 8b11a63..9b068b7 100644
--- a/sound/soc/codecs/florida.c
+++ b/sound/soc/codecs/florida.c
@@ -215,6 +215,8 @@ static int florida_sysclk_ev(struct snd_soc_dapm_widget *w,
 		break;
 	}
 
+	msleep(10);
+
 	return 0;
 }
 
-- 
1.7.9.5

